src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/MicrosoftExtensionsLogging/MicrosoftLoggingWrapper.cs

```diff
@@ -53,52 +55,64 @@ private void RecordLogMessage(MethodCall methodCall, MEL.ILogger logger, IAgent
                 Func<object, string> getLevelFunc = mc => ((MethodCall)mc).MethodArguments[0].ToString();
                 Func<object, string> getRenderedMessageFunc = mc => ((MethodCall)mc).MethodArguments[2].ToString();
                 Func<object, Exception> getLogExceptionFunc = mc => ((MethodCall)mc).MethodArguments[3] as Exception; // using "as" since we want a null if missing
-                Func<object, Dictionary<string, object>> getContextDataFunc = nothx => GetContextData(logger);
+                Func<object, Dictionary<string, object>> getContextDataFunc = nothx => GetContextData(logger, agent);
 
                 var xapi = agent.GetExperimentalApi();
                 xapi.RecordLogMessage(WrapperName, methodCall, getTimestampFunc, getLevelFunc, getRenderedMessageFunc, getLogExceptionFunc, getContextDataFunc, agent.TraceMetadata.SpanId, agent.TraceMetadata.TraceId);
             }
         }
 
-        private static Dictionary<string, object> GetContextData(MEL.ILogger logger)
+        private static Dictionary<string, object> GetContextData(MEL.ILogger logger, IAgent agent)
         {
-            // We are trying to access this property:
-            // logger.Loggers[0].Logger.ScopeProvider
+            if (_contextDataNotSupported) // short circuit if we previously got an exception trying to access context data
+                return null;
```

Comment:
Pedantic feedback.. our coding standard is supposed to include `{ }` even for one line control logic

Reply:
Thanks - first time I'm hearing there's a coding standard. Looked around and found it in the team wiki (where I probably should have noticed it before), but didn't see anything specifically addressing the need for curly braces in this case. I've added a bullet point in the WB Design Meeting for this week to discuss updating the coding standards.